### ISSUE TITLE
Print representation of structs in Emacs 26 breaks tests

### ISSUE THREAD

**Issue by xiongtx** [clojure-emacs/cider#2133]
Some tests are [failing when run against Emacs 26](https://travis-ci.org/clojure-emacs/cider/jobs/314678704#L1868) due to the new `#s` struct print syntax, which seems to be due to the use of [records to represent structs](http://git.savannah.gnu.org/cgit/emacs.git/tree/doc/lispref/records.texi#n27).

**Comment by gonewest818** [comment]
I'm curious about this one:

```
FAILED: Expected `(nrepl-bdecode "lli1ei2ei3eeli5ei6eee")' to be `equal' to 
`([cl-struct-queue nil nil] . [cl-struct-nrepl-response-queue (((1 2 3) (5 6))) (((1 2 3) (5 6))) nil])', 
but instead it was `(#s(queue nil nil) . #s(nrepl-response-queue (((1 2 3) (5 6))) (((1 2 3) (5 6))) nil))'
which does not match because: (car (different-types #s(queue nil nil) [cl-struct-queue nil nil])).
```

It doesn't look like the reader supports `#s` syntax for arbitrary structs, so how does one specify the correct match for that test?

**Comment by xiongtx** [comment]
The Emacs 26 reader_should_ support `#s` for structs. Is that not the case?

**Comment by gonewest818** [comment]
I may be mistaken. I saw documentation somewhere that said that #s notation was supported only for hash-tables. I didn’t understand why that would be the case, but the way it was written it seemed like they were going out of their way to make the point.

**Comment by xiongtx** [comment]
>  `#s` notation was supported only for hash-tables

Funny you should mention this--I actually [asked about it](https://lists.gnu.org/archive/html/emacs-devel/2017-12/msg00320.html) on `emacs-devel`. Seems the decision by Stefan Monnier to use `#s` (which corresponds to [`#s` for structures in CL](https://www.cs.cmu.edu/Groups/AI/html/cltl/clm/node191.html)) was a controversial one. 

Ultimately I think Stefan made the right choice--closer alignment w/ CL is a good thing.

**Comment by gonewest818** [comment]
Ok this works in emacs 26.0.90

```
 (expect (nrepl-bdecode "lli1ei2ei3eeli5ei6eee")
                  :to-equal (cons
                             #s(queue nil nil)
                             #s(nrepl-response-queue (((1 2 3) (5 6))) (((1 2 3) (5 6))) nil)))))
```

But it’s not valid code for older versions of emacs, so presumably a macro or something is needed to conditionally compile this new test canonical only when the emacs version is 26 or greater.

**Comment by xiongtx** [comment]
How about [constructing the struct](https://www.gnu.org/software/emacs/manual/html_node/cl/Structures.html#Structures) directly via `make-struct`? E.g. ` (make-struct queue nil nil)`. That should get around the issue w/ literals.

**Comment by xiongtx** [comment]
[Building the struct on-the-fly](https://github.com/clojure-emacs/cider/issues/2133#issuecomment-351924062) seems to be the best way to avoid this read syntax problem.
